### DISCUSSION TITLE
Create model custom attributes in documentation?

### DISCUSSION THREAD

**Discussion by milenmk** [laravel/framework#54044] [Ideas] [answered]
Hi all.

Correct me if I am wrong, but nowhere in the documentation is it explained how to cast/create a custom attribute for your model, that is not part of the fillable properties. What I mean is, let's have a user model with 2 fields: name and last_name.

Now, anywhere on your code, you can do like `$user->name .' '. $user->last_name`. However, you can also do this:

```
// In the User model

    public function getFullNameAttribute(): string
    {
        return $this->name). ' '. $this->last_name;
    }
```

and then just use `$user->full_name`. 

may be it should be added somewhere in the documentation, as this is very usefull.

**Comment by ghost** [comment, marked as answer]
They moved away from this type of mutators in Laravel 9. https://laravel.com/docs/11.x/eloquent-mutators#defining-a-mutator vs https://laravel.com/docs/8.x/eloquent-mutators#defining-a-mutator

**Reply by milenmk** [reply]
Do you mean I should move to this new way, as the one I am using may become unusable in future versions?

**Reply by ghost** [reply]
I didn't move away from it. They stated it is still usable in the old way.

**Reply by ghost** [reply]
But this second way of doing it, restricts the user from defining a relation with the same name as the mutator.

**Reply by ghost** [reply]
see this https://github.com/laravel/framework/discussions/54114
Old way is the best way.
